python/src/iceberg/schema.py

```diff
@@ -290,6 +292,8 @@ def _(obj: StructType, visitor: SchemaVisitor[T]) -> T:
         visitor.before_field(field)
         try:
             result = visit(field.type, visitor)
+        except Exception:
+            logger.exception("Unable to visit the field in the schema")
```

Comment:
Should this also log which field wasn't handled? Something like `f"Unable to visit field: {field}"`?

Reply:
I've changed the code to re-raise the exception. I had a couple of time during debugging that it silently swallowed the exception, which didn't really help in the process of debugging :)